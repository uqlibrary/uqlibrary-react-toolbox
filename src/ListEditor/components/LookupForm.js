import React, {Component} from 'react';
import PropTypes from 'prop-types';

export default class LookupForm extends Component {
    static propTypes = {
        onAdd: PropTypes.func.isRequired,
        locale: PropTypes.object,
        disabled: PropTypes.bool,
        inputField: PropTypes.func,
        errorText: PropTypes.string
    };

    static defaultProps = {
        isValid: () => '',
        locale: {
            inputFieldLabel: 'Item name',
            inputFieldHint: 'Please type the item name, then select from the list',
        }
    }

    constructor(props) {
        super(props);
    }

    addKeyValueItem = (item) => {
        this.props.onAdd(item);
    }

    render() {
        console.log(this.props.errorText);
        return (
            <div className="columns">
                <div className="column">
                    {
                        this.props.inputField &&
                        <this.props.inputField
                            input={{onChange: this.props.onAdd}}
                            floatingLabelText={this.props.locale.inputFieldLabel}
                            hintText={this.props.locale.inputFieldHint}
                            disabled={this.props.disabled}
                            errorText={this.props.errorText} />
                    }
                </div>
            </div>
        );
    }
}
