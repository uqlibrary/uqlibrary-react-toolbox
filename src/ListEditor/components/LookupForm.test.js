import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import LookupForm from './LookupForm';
import injectTapEventPlugin from 'react-tap-event-plugin';

beforeAll(() => {
    injectTapEventPlugin();
});

function setup({onAdd, isValid, disabled}){

    const props = {
        inputField: () => (<span />),
        onAdd: onAdd || jest.fn(), // : PropTypes.func.isRequired,
        isValid: isValid || jest.fn(() => ('')), // PropTypes.func,
        disabled // : PropTypes.bool
        //locale, // : PropTypes.object,
    };
    return shallow(<LookupForm {...props} />);
}

describe('LookupForm tests ', () => {
    it('should render lookup form', () => {
        const wrapper = setup({ });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
