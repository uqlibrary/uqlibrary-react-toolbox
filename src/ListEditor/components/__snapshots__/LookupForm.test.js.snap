// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LookupForm tests  should render lookup form 1`] = `
<div
  className="columns"
>
  <div
    className="column"
  >
    <inputField
      floatingLabelText="Item name"
      hintText="Please type the item name, then select from the list"
      input={
        Object {
          "onChange": [Function],
        }
      }
    />
  </div>
</div>
`;
